modules/index-service/src/main/java/org/opencastproject/index/service/impl/IndexServiceImpl.java

```diff
@@ -1363,6 +1391,88 @@ public MetadataList updateEventMetadata(String id, MetadataList metadataList, El
     return metadataList;
   }
 
+  /**
+   * Updates the publications of a mediapackage with new metadata and access rights.
+   * @param mediaPackage
+   */
+  private void updatePublication(MediaPackage mediaPackage) {
+    for (Publication publication : mediaPackage.getPublications()) {
+      String channelId = publication.getChannel();
+      String publicationId = publication.getIdentifier();
+      URI publicationURI = publication.getURI();
+      MimeType publicationMimeType = publication.getMimeType();
+
+      if (channelId == null || publicationURI == null || publicationMimeType == null || publicationId == null) {
+        // No publication, do nothing
+        continue;
+      }
+
+      // Get metadata and acls
+      SimpleElementSelector elementSelector = new SimpleElementSelector();
+      // This does not necessarily work for extended metadata, which can have completely different flavors.
+//      elementSelector.addFlavor("dublincore/*");
+      // Relies on ACLs having this particular flavor
+      elementSelector.addFlavor("security/*");
+      Collection<MediaPackageElement> elements = elementSelector.select(mediaPackage, false);
+
+      Set<String> elementIds = new HashSet<String>();
+      for (MediaPackageElement elem : elements) {
+        elementIds.add(elem.getIdentifier());
+      }
+
+      // To make sure we hit all extended metadata catalogs, let's just get all catalogs
+      for (Catalog catalog : mediaPackage.getCatalogs()) {
+        elementIds.add(catalog.getIdentifier());
+      }
+
+      if (elementIds.size() < 1) {
+        // Nothing to republish, do nothing
+        continue;
+      }
+
+      // Remove publication from mediapackage
+      for (Publication publicationElement : mediaPackage.getPublications()) {
+        if (channelId.equals(publicationElement.getChannel())) {
+          mediaPackage.remove(publicationElement);
+        }
+      }
+
+      List<MediaPackageElement> downloadElements = new ArrayList<>();
+      List<MediaPackageElement> streamingElements = new ArrayList<>();
+      try {
+        downloadElements = downloadDistributionService.distributeSync(channelId, mediaPackage, elementIds, false);
+        if (streamingDistributionService != null && streamingDistributionService.publishToStreaming()) {
+          streamingElements = streamingDistributionService.distributeSync(channelId, mediaPackage, elementIds);
+        }
+      } catch (DistributionException e) {
+        throw new RuntimeException(e);
+      }
+      logger.debug("Distribute of mediapackage {} to channel {} completed", mediaPackage, channelId);
+
+      // Re-add publication
+      Publication newPublication = PublicationImpl.publication(publicationId, channelId, publicationURI, publicationMimeType);
```

Comment:
Is it okay to reuse the publication ID? 🤔

Reply:
Hopefully. It's definitely easier than generating a new one and having to figure out how to generate a publication uri with the new id for each publication.